javascripts/custom-navigation/connectors/discovery-above/category-sidebar-outlet.js

```diff
@@ -0,0 +1,20 @@
+import discourseComputed from "discourse-common/utils/decorators";
+import { inject as service } from "@ember/service";
+
+export default {
+  setupComponent() {
+    this.reopen({
+      router: service(),
+
+      @discourseComputed("router.currentRouteName")
+      showOnRoute(currentRouteName) {
+        console.log(currentRouteName);
+        if (currentRouteName.indexOf("discovery") > -1 || currentRouteName.indexOf("tag") > -1) {
+          return true;
+        } else {
+          return false;
+        }
+      }
+    });
+  }
+};
```

Comment:
This can be simplified:

```suggestion
import { getOwner } from "discourse-common/lib/get-owner";

export default {
  shouldRender(args, component) {
    const router = getOwner(this).lookup("router:main");
    return router.currentRouteName.includes("discovery") || router.currentRouteName.includes("tag");
  },
};
```

then the template does not need to be changed.

---

But on another note: those discovery/tag checks are not ideal, as they can match unrelated paths especially if new ones are added by plugins/theme components (e.g. `"stage".includes("tag")`)

---

(Both things apply to the other outlet too)

Reply:
> This can be simplified:
> 
> then the template does not need to be changed.
> 
> But on another note: those discovery/tag checks are not ideal, as they can match unrelated paths especially if new ones are added by plugins/theme components (e.g. `"stage".includes("tag")`)
> 
> (Both things apply to the other outlet too)

Will push the changes to the other route. Do you have a suggestion on what checks I should use instead of these?

Reply:
Hmm, probably something like `/^discovery([A-Z]|-|$)/.test(name)`